eessi/testsuite/hooks.py

```diff
@@ -489,6 +489,20 @@ def filter_valid_systems_by_device_type(test: rfm.RegressionTest, required_devic
     log(f'valid_systems set to {test.valid_systems}')
 
 
+def filter_valid_systems_for_offline_partitions(test: rfm.RegressionTest):
+    """
+    Filter tests if they require internet access to run.
+    Filtering is done using features, i.e. when offline is requested as a feature.
+    Any partition that includes this feature in the ReFrame configuration file will effectively be filtered out.
+    """
+    valid_systems = r'-offline'
```

Comment:
can you add this to the `_Features` class in `constants.py`?
```python
OFFLINE: str = 'offline'
```
and then you can replace this line with
 
```suggestion
    valid_systems = f'-{FEATURES.OFFLINE}'
```